src/test/java/com/github/tomakehurst/wiremock/matching/EqualToXmlPatternTest.java

```diff
@@ -450,7 +450,8 @@ public void namespaceComparisonCanBeExcluded() {
             + "    <st:thing>Match this</st:thing>\n"
             + "</stuff>";
 
-    MatchResult matchResult = equalToXml(expected).match(actual);
+    MatchResult matchResult =
```

Comment:
I think fixing this test and enabling the namespace flag in order to make it pass again is the right thing to do, but it should be in a separate PR. I suggest you create one for that first, we'll get it merged then you can slim this one down so it just fixes the comments performance issue.

Reply:
Just adding the namespace flag causes quite a few tests to fail so I think we need to examine these carefully to see if we need to preserve old behaviour in at least some cases.

Reply:
ive put the Xml refactor and namespace aware change into a separate pr here https://github.com/wiremock/wiremock/pull/2945